server/middlewares/validate.js

```diff
@@ -0,0 +1,66 @@
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const validation = {
+  validateRegistrationEntry(request, response, next) {
+    const required = ['email', 'password', 'firstName', 'lastName', 'phoneNumber', 'confirmPassword'];
+    const collection = request.body;
+    let isValid = true;
+    const errors = {};
+    for (let i = 0; i < required.length; i += 1) {
+      if (!collection[required[i]]) {
+        isValid = false;
+        errors[required[i]] = `please provide ${required[i]}`;
+      }
+    }
+    if (collection.email) {
+      if ((validation.isEmail(collection.email) === false)) {
+        isValid = false;
+        errors.email = 'please provide a valid email address';
+      }
+    }
+    if (collection.password && !collection.password.replace(/\s/g, '').length) {
+      errors.password = 'Please enter a valid password';
+      isValid = false;
+    }
```

Comment:
This block of code can be abstracted into a helper function that can be imported here. Kindly look into it. Reach out if you need help.